### ISSUE TITLE
Tool spanning and junction read weights are 0 in random forest (issue in 1.3.6, not in 1.3.4; solution proposed)

### ISSUE THREAD

**Issue by jrouhana** [TRON-Bioinformatics/EasyFuse#24]
I initially [submitted a pull-request in June](https://github.com/TRON-Bioinformatics/EasyFuse/pull/11) regarding the Docker container being built on Python2 but using int-division in count normalization. This was not corrected in the 1.3.6 update. I've gone ahead and updated the pull request to include 2 other instances where int-division is used improperly and causes issues in Python2, which is the default version of Python in the Docker image that you provide.

As of the 1.3.6 update (1.3.5 untested, but I believe the issue was introduced there), it seems that the random forest provided with your Docker image has been trained with the problematic int division Docker image. Checking feature importance from `randomForest` highlights this:

```
> importance(model)
                       MeanDecreaseGini
type                         73.7250814
exon_boundary                41.0677487
fusioncatcher_detected        0.5083960
infusion_detected             0.9841079
mapsplice_detected           10.8953450
soapfuse_detected            13.9531531
starfusion_detected           6.3734725
fusioncatcher_junc            0.0000000
infusion_junc                 0.0000000
mapsplice_junc                0.0000000
soapfuse_junc                 0.0000000
starfusion_junc               0.0000000
fusioncatcher_span            0.0000000
infusion_span                 0.0000000
mapsplice_span                0.0000000
soapfuse_span                 0.0000000
starfusion_span               0.0000000
tool_frac                    28.4812380
ft_a_best                    30.9791804
ft_b_best                    27.9950978
ft_junc_best                 36.5403701
wt1_junc_best                28.2912068
wt2_junc_best                24.8001392
ft_span_best                 64.0483935
wt1_span_best                32.1409714
wt2_span_best                24.5208664
```

It is apparent that the spanning and junction counts were rounded to 0 before training. This is not the case in 1.3.4:
```
                       MeanDecreaseGini
type                         63.9149879
exon_boundary                34.9394928
fusioncatcher_detected       12.6583086
infusion_detected             0.4688008
mapsplice_detected            2.6176553
soapfuse_detected             3.4534515
starfusion_detected           2.6727990
fusioncatcher_junc           20.4798514
infusion_junc                 0.7681182
mapsplice_junc                3.6025693
soapfuse_junc                18.2990984
starfusion_junc              12.0836191
fusioncatcher_span           13.5411700
infusion_span                 0.1305807
mapsplice_span                4.8624121
soapfuse_span                30.8595291
starfusion_span              31.9575740
tool_frac                    14.4295199
ft_a_best                    16.9865021
ft_b_best                    15.8091136
ft_junc_best                 18.0476760
wt1_junc_best                17.5306386
wt2_junc_best                13.9102512
ft_span_best                 53.1208341
wt1_span_best                18.6558268
wt2_span_best                14.7943119
```

I believe the pull request I submitted will resolve this issue (but a new model will need to be trained).

**Comment by ibn-salem** [comment]
Many thanks for reporting this issue. As you noted this is related to #21 We are currently looking into this and want to ensure that python3 is used in the Docker image and will solve the floating point issue. This will be available in the next release.

**Comment by jrouhana** [comment]
Thanks for responding @ibn-salem. Do you guys have an estimated timeline for the next release?

**Comment by ibn-salem** [comment]
This is fixed in version 1.3.7
